fix: fix email validation

andresmoschini opened the pull request:
Hi team, 

I found that the last changes in email validation do not allow emails with only a letter before @, I fixed that and added some tests.

Could you review?

CC: @CarlosSampedro, maybe you could think in other scenarios.
CC: @lucianaDilizio, if I am not wrong, you have applied the previous change in another place, you should consider also apply this one.

cbernat commented on src/validations.test.js:
```diff
@@ -52,6 +52,61 @@ describe('validations', () => {
       expect(result).toBeNull();
     });
 
+    it('should not accept an email address with dot (.) before @', () => {
```
What do you think about adding a test with an email like `email@mail.com.ar`?

lucianaDilizio commented:
This is the PR with the same changes in Automation: https://github.com/MakingSense/Doppler/pull/5944

CarlosSampedro commented:
Hi @andresmoschini ,

I have a question regarding the email validation:
Should this email validation be applied just un logn in/ Sign up or should it be applied with subscribers too?

On the other hand, I have verified the regular expression you made with the email validation (sign up) in Production, and I have  detected differencies with some chars. The following emails are refused by the regular expression but they are accepted in production:
T&P@gmail.com.ar
T/P@gmail.com.ar
T~P@gmail.com.ar

Later I will take a look the Unit Test escenarios looking for a new ones.

Regards

andresmoschini commented:
Buen trabajo @CarlosSampedro, según estuvimos chateando, deberíamos haber utilizado la misma expresión regular que se está usando en Doppler.

Por ahora, vamos a investigar si hay usuarios con caracteres no soportados, y en ese caso corregiremos la expresión regular actual. 

Luego, con más tiempo intentaremos unificar ambas validaciones.

andresmoschini commented:
@CarlosSampedro @lucianaDilizio I have updated it to support some weird real scenarios and simplified the regular expression.

BTW, thanks @fgchaio for the research!

andresmoschini commented:
:tada: This PR is included in version 1.0.0 :tada:

The release is available on [GitHub release](https://github.com/FromDoppler/doppler-webapp/releases/tag/v1.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
